packages/react-core/src/components/Button/Button.js

```diff
@@ -65,9 +70,11 @@ const Button = ({
   variant,
   ...props
 }) => (
-  <button
+  <Component
     {...props}
     disabled={isDisabled}
+    aria-disabled={isDisabled && Component !== 'button' ? true : null}
+    tabIndex={isDisabled && Component !== 'button' ? -1 : null}
```

Comment:
Similar to the logic you have here for links, the `disabled` attribute should also have logic to not be included if it's a link.